### ISSUE TITLE
Maybe uninitialized compiler warning

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#8233]
### Description of the enhancement or error report
gcc on linux gives a bunch of compile time warnings in dbg mode about potentially ininitialized variables. Now if those were real issues our valgrind testing should find them. Nonetheless I suggest cleaning those up (by explicitly initializing them) to remove these warnings.

### Rationale for the enhancement or information for reproducing the error
dbg should compile without _any_ warnings as not to obfuscate real new issues.

### Identified impact
Cleaner compiler output

## Warnings:
```
/home/daniel/Programs/INL/moose/framework/src/base/ComputeResidualThread.C: In member function ‘virtual void ComputeResidualThread::onElement(const libMesh::Elem*)’:
/home/daniel/Programs/INL/moose/framework/src/base/ComputeResidualThread.C:92:44: warning: ‘warehouse’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   const MooseObjectWarehouse<KernelBase> * warehouse;
                                            ^~~~~~~~~
```

```
/home/daniel/Programs/INL/moose/framework/src/transfers/MultiAppNearestNodeTransfer.C: In member function ‘virtual void MultiAppNearestNodeTransfer::execute()’:
/home/daniel/Programs/INL/moose/framework/src/transfers/MultiAppNearestNodeTransfer.C:480:22: warning: ‘solution’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     solution->close();
                      ^
/home/daniel/Programs/INL/moose/framework/src/transfers/MultiAppNearestNodeTransfer.C:356:27: note: ‘solution’ was declared here
     NumericVector<Real> * solution;
                           ^~~~~~~~
```

```
/home/daniel/Programs/INL/moose/modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticTensileMulti.C: In member function ‘virtual bool TensorMechanicsPlasticTensileMulti::doReturnMap(const RankTwoTensor&, libMesh::Real, const RankFourTensor&, libMesh::Real, RankTwoTensor&, libMesh::Real&, std::__debug::vector<double>&, RankTwoTensor&, std::__debug::vector<double>&, bool&) const’:
/home/daniel/Programs/INL/moose/modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticTensileMulti.C:288:22: warning: ‘nr_converged’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     if (nr_converged && KuhnTuckerOK(returned_stress, dpm, str, ep_plastic_tolerance))
         ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/daniel/Programs/INL/moose/modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticTensileMulti.C:272:8: note: ‘nr_converged’ was declared here
   bool nr_converged;
        ^~~~~~~~~~~~
```

```
/home/daniel/Programs/INL/moose/modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticMohrCoulombMulti.C: In member function ‘bool TensorMechanicsPlasticMohrCoulombMulti::doReturnMap(const RankTwoTensor&, libMesh::Real, const RankFourTensor&, libMesh::Real, RankTwoTensor&, libMesh::Real&, std::__debug::vector<double>&, RankTwoTensor&, std::__debug::vector<double>&, bool&) const’:
/home/daniel/Programs/INL/moose/modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticMohrCoulombMulti.C:427:22: warning: ‘kt_success’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     if (nr_converged && kt_success)
         ~~~~~~~~~~~~~^~~~~~~~~~~~~
/home/daniel/Programs/INL/moose/modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticMohrCoulombMulti.C:377:8: note: ‘kt_success’ was declared here
   bool kt_success;
        ^~~~~~~~~~
```

```
/home/daniel/Programs/INL/moose/modules/solid_mechanics/src/actions/DomainIntegralAction.C:678:31: warning: ‘i_kiii’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       sif_mode_enum_vec[i_kiii] = "KIII";
                               ^
/home/daniel/Programs/INL/moose/modules/solid_mechanics/src/actions/DomainIntegralAction.C:676:30: warning: ‘i_kii’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       sif_mode_enum_vec[i_kii] = "KII";
                              ^
/home/daniel/Programs/INL/moose/modules/solid_mechanics/src/actions/DomainIntegralAction.C:680:28: warning: ‘i_t’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       sif_mode_enum_vec[i_t] = "T";
                            ^
```

```
/home/daniel/Programs/INL/moose/modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C: In member function ‘void CrackFrontDefinition::pickLoopCrackEndNodes(std::__debug::vector<unsigned int>&, std::__debug::set<unsigned int>&, std::__debug::map<unsigned int, std::__debug::vector<unsigned int> >&, std::__debug::vector<std::__debug::vector<unsigned int> >&)’:
/home/daniel/Programs/INL/moose/modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C:464:14: warning: ‘max_dist_node’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     end_node = max_dist_node;
     ~~~~~~~~~^~~~~~~~~~~~~~~
```

**Comment by dschwen** [comment]
There are also lots of unused parameter warnings in the rdg module @yidongxiainl

**Comment by dschwen** [comment]
And another one:

```
/home/daniel/Programs/INL/moose/modules/phase_field/src/postprocessors/GrainTracker.C: In member function ‘void GrainTracker::trackGrains()’:
/home/daniel/Programs/INL/moose/modules/phase_field/src/postprocessors/GrainTracker.C:631:72: warning: ‘closest_match_index’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             new_grain_index_to_existing_grain_index[closest_match_index] = old_grain_index;
                                                                        ^
```

Btw. I mentioned that one GT test crashes on my mac?

**Comment by dschwen** [comment]
And two more:

```
/home/daniel/Programs/INL/moose/modules/chemical_reactions/src/kernels/CoupledDiffusionReactionSub.C: In member function ‘virtual libMesh::Real CoupledDiffusionReactionSub::computeQpOffDiagJacobian(unsigned int)’:
/home/daniel/Programs/INL/moose/modules/chemical_reactions/src/kernels/CoupledDiffusionReactionSub.C:136:24: warning: ‘val_jvar’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       diff3 = val_jvar * _sto_v[i] * std::pow((*_vals[i])[_qp], _sto_v[i] - 1.0) * (*_grad_vals[i])[_qp];
```

```
/home/daniel/Programs/INL/moose/modules/chemical_reactions/src/kernels/CoupledConvectionReactionSub.C: In member function ‘virtual libMesh::Real CoupledConvectionReactionSub::computeQpOffDiagJacobian(unsigned int)’:
/home/daniel/Programs/INL/moose/modules/chemical_reactions/src/kernels/CoupledConvectionReactionSub.C:138:24: warning: ‘val_jvar’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       diff3 = val_jvar * _sto_v[i] * std::pow((*_vals[i])[_qp], _sto_v[i] - 1.0) * (*_grad_vals[i])[_qp];
```

**Comment by yidongxiainl** [comment]
Thanks, @dschwen I will check rdg module and try to remove as many warnings as possible.

**Comment by yidongxiainl** [comment]
@dschwen  Some of the "unused parameter" warnings in the rdg module will not be removed, because they are part of standard template for flux and boundary condition functions. For example, in some type of flux functions, those parameters are not used; but in some others, they are needed.

**Comment by dschwen** [comment]
@yidongxiainl and I talked at lunch about this. Just to have a record of it: The warnings can be squashed by commenting out the parameter names (just leaving in the types). See 11f0875 for example (that commit also shows the `libmesh_dbg_var` macro, which is useful for variables that only get used in asserts).

**Comment by yidongxiainl** [comment]
@dschwen  I will grab your branch "uninit_8233" and remove the warnings in rdg module.

**Comment by yidongxiainl** [comment]
@dschwen can I push commits to your "uninit_8233" branch once I'm done with my part?

**Comment by dschwen** [comment]
No, but you can make a PR to my repo. I suggest making the changes in an independent branch though.

**Comment by dschwen** [comment]
I think I fixed some already. I can do the rest as well if any are left.

**Comment by permcody** [comment]
@dschwen - which GT test crashes for you?

**Comment by dschwen** [comment]
git cleaned everything and now it seems fine. There was some update foo bar going on before the break.

**Comment by permcody** [comment]
This issue has been completely resolved by #8290 and earlier merges.
